docs/contributing/docs-and-website-components.md

```diff
@@ -0,0 +1,418 @@
+
+It is used like this:
+
+```markdown:title=blog/new-announcement.md
+---
+title: Big News!
+author: Jay Gatsby
+---
+
+<!-- highlight-start -->
+<Pullquote citation="Winston Churchill" narrow={true}>
+  To improve is to change, so to be perfect is to have changed often.
+</Pullquote>
+<!-- highlight-end -->
+```
+
+#### Sample
+
+Rendered, the component looks like this:
+
+<Pullquote narrow={true} citation="Winston Churchill">
+  To improve is to change, so to be perfect is to have changed often.
+</Pullquote>
+
+---
+
+## Other available components
+
+Other less commonly used components aren't globally available, but can imported at the top of a Markdown file and used in other docs.
+
+---
+
+### Layer Model
+
+The `<LayerModel />` was made to help explain concepts of how Gatsby works at a high level. It conceptually breaks Gatsby into different layers and shows how data is pulled, aggregated, and eventually rendered as an app. It's used on docs pages to help explain how Gatsby works at different levels.
+
+#### Usage
+
+The Layer Model component takes one prop:
+
+- `initialLayer` - defaults to `Content`, can be one of `Content`, `Build`, `Data`, `View`, `App` that correspond to the different layers
+
+```markdown
+---
+title: GraphQL Concepts in Gatsby
+---
+
+import LayerModel from "../../www/src/components/layer-model"
+
+To help understand how GraphQL works in Gatsby...
+
+<LayerModel initialLayer="Data" />
+```
+
+#### Sample
+
+When used, it looks like this:
+
+<LayerModel initialLayer="Data" />
+
+### Horizontal Navigation List
+
+The `<HorizontalNavList />` was made for the [Glossary](/docs/glossary/), and renders a list of links to alphabetical subheadings on the page in a horizontal format. Because of its specific use case, it isn't made globally available but can be imported if needed on other pages.
+
+#### Usage
+
+The Horizontal Nav List component takes two props:
+
+- `slug` - which is provided in the props of the page by default
+- `items` - an array of strings for items to render and wrap with a `<Link />` to subheadings
+
+The docs on Gatsbyjs.org use the [gatsby-remark-autolink-headers](/packages/gatsby-remark-autolink-headers/) plugin to automatically apply hover links to heading tags across docs pages. Because it automatically creates links to subheadings on pages like the glossary, the Horizontal Nav List can supply matching links (like `"guide-list"` which would align with the automatically created link at `/docs/docs-and-website-components#guide-list`).
+
+<!-- prettier-ignore -->
+```markdown
+---
+title: Glossary
+---
+
+import HorizontalNavList from "../../www/src/components/horizontal-nav-list.js"
+
+The glossary defines key vocabulary...
+
+---
+
+<HorizontalNavList
+  slug={props.slug}
+  items={["guide-list", "egghead-embed", "pull-quote", "layer-model", "horizontal-navigation-list"]}
+/>
+```
+
```

Comment:
maybe also nice to see here the glossary Nav List?

https://github.com/gatsbyjs/gatsby/blob/d545b647f3291b15e1e1693adfe6064a36bc357d/docs/docs/glossary.md#L10-L13

Reply:
That's what I thought initially, and we certainly could....but the existing usage is more consistent with the current page as the glossary one wouldn't link to any relevant headings.

Reply:
yes, but maybe as an small note about the split ;) 

```markdown
An alphabet glossary can be generated with `items={"ABCDEFGHIJKLMNOPQRSTUVWXYZ".split("")}`
```

Reply:
I thought about that too but felt like it made it a little harder to see what the prop was doing. If someone looks at the component on the glossary page they can still find that way too use it, but it's just a bit trickier of a way to see how the prop is used since they have to know what `split` does with this example.